kyo-prelude/shared/src/main/scala/kyo/Stream.scala

```diff
@@ -352,6 +354,85 @@ sealed abstract class Stream[V, -S]:
                     else Emit.valueWith(c)(((), cont(())))
         ))
 
+    /** Transform the stream with a partial function, filtering out values for which the partial function is undefined. Combines the
+      * functionality of map and filter.
+      *
+      * @param f
+      *   Partial function transforming V to V2
+      * @return
+      *   A new stream containing transformed elements
+      */
+    def collect[V2, S2](f: V => Maybe[V2] < S2)(using
+        tag: Tag[Emit[Chunk[V]]],
+        t2: Tag[Emit[Chunk[V2]]],
+        frame: Frame
+    ): Stream[V2, S & S2] =
+        Stream[V2, S & S2](ArrowEffect.handleState(tag, (), emit)(
+            [C] =>
+                (input, _, cont) =>
```

Comment:
This pattern of using `handleState` without a state so it's possible to suspend other effects during handling is common. I'll follow up on a kernel improvement.